src/db.js

```diff
@@ -0,0 +1,273 @@
+    result = {
+        address: address,
+        norequests: 1,
+        dailycount: 0,
+        weeklycount: 0,
+        firstrequesttime: now,
+        dailytime: now,
+        weeklytime: now,
+        validatedTx: "",
+        unaccountedamount: 0,
+        unaccountedamount: ""
+    }
+    return result;
+}
+
+async function checkDailyLimit(addressDetails){
+    const dailycount = await resetDailyCount(addressDetails);
+    console.log(dailycount);
+    return dailycount <= dailyLimit;
+
+}
+
+async function resetDailyCount(addressDetails){
+    const now = new Date();
+    // console.log(addressDetails);
+    const address = addressDetails.address;
+    const dailytime = addressDetails.dailytime;
+    if ((Math.floor(now.getTime()/1000 - Math.floor(dailytime.getTime()/1000))) > 86400){
+        //update
+        console.log('Resetting...');
+        const update = 'update depositortest set dailycount=0,dailytime=$1 where address= $2 returning dailycount'
+        const values = [now,address]
+        const dailycount = await pool.query(update,values);
+        return dailycount.rows[0].dailycount; //daily limit has been reset
+    }
+    return addressDetails.dailycount;
+    
+
+}
+
+async function checkWeeklyLimit(addressDetails){
+    const weeklycount = await resetWeeklyCount(addressDetails);
+    return weeklycount <= weeklyLimit;
+}
+
+async function resetWeeklyCount(addressDetails){
+    const now = new Date();
+    const address = addressDetails.address;
+    const weeklytime = addressDetails.weeklytime;
+
+    if ((Math.floor(now.getTime()/1000 - Math.floor(weeklytime.getTime()/1000))) > 604800){
+        //update
+        const update = 'update depositortest set weeklycount=0,weeklytime=$1 where address= $2 returning weeklycount'
+        const values = [now,address]
+        const weeklycount = await pool.query(update,values);
+        return weeklycount.rows[0].weeklycount; //weekly limit has been reset
+    }
+    return addressDetails.weeklycount;
+}
+
+async function resetNoRequests(addressDetails){
+    // console.log(addressDetails)
+    const now = new Date();
+    const address = addressDetails.address;
+    const firstrequesttime = addressDetails.firstrequesttime;
+    if ((Math.floor(now.getTime()/1000 - Math.floor(firstrequesttime.getTime()/1000))) > 172800){
+        //update
+        const update = 'update depositortest set norequests=0,firstrequesttime=$1 where address= $2 returning norequests'
+        const values = [now,address]
+        const norequests = await pool.query(update,values);
+        return norequests.rows[0].norequests; //daily limit has been reset
+    }
+    return addressDetails.norequests;
+}
+
+async function updateCounts(addressDetails,topUpAmount){
+    var newDailyCount = addressDetails.dailycount + topUpAmount;
+    var newWeeklyCount = addressDetails.weeklycount + topUpAmount;
+    
+    const update = 'update depositortest set dailycount= $1,weeklycount= $2 where address= $3';
```

Comment:
`address` seems not escaped, so it can be possible to run SQL Injection

Reply:
Also check if weeklycount comes from outside of system and escape it by wrapping with `integer` type